### PR TITLE
PR from Zhuo

### PR DESCRIPTION
**Pull request by Zhangdroid**
## Overview
* Prettier, eslint, commitlint
* 100% TypeScript
* GitHub actions for CI 
* Auto update dependency by renovate
* Dark mode

### Frontend
* React, React-Router
* Styled Components, polished
* Immer, swr

### Backend
* Express, Prisma 2

### TODO
* Pagenation
* Auto gen types
* Auto deploy

### CODE REVIEW

File: web/src/pages/Home/index.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  }
+  return (
+    <Container>
+      <TalkList
+        data={orderBy(data, ["vote", "title"], ["desc", "asc"])}
+        onVote={async item => {
+          if (!user) {
+            history.push(LOGIN, {
+              from: `/?vote=${item.id}`
+            });
```
[the remark this answers is not in the document]
Comment (Zhangdroid) [reply web/src/pages/Home/index.tsx]: Yes, I wanted bring this state to the login page and bring it back to home, then automatically vote for user after login, but not sure if it's necessary. Should leave a comment here 😂

File: api/package.json [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+          }
+        }
+      ],
+      "@babel/preset-react",
+      "@babel/preset-typescript"
+    ],
+    "plugins": [
+      "@babel/proposal-optional-chaining"
+    ]
+  },
```
[the remark this answers is not in the document]
Comment (Zhangdroid) [reply api/package.json]: Yeah, I'm using `tsc` to compile source, but the tests(jest) using babel to compile so I need it. About `@babel/proposal-optional-chaining`, I think it's very useful but still not in ECMAScript standard(but in Stage 4), it should can be removed very soon: https://github.com/babel/babel/issues/10809

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+      matrix:
+        node-version: [10.x, 12.x]
+
+    steps:
+      - uses: actions/checkout@v1
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+      - run: npm install -g yarn
```
[the remark this answers is not in the document]
Comment (Zhangdroid) [reply .github/workflows/ci.yaml]: Thanks for head up, fixed it :)

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [10.x, 12.x]
+
+    steps:
+      - uses: actions/checkout@v1
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
```
[the remark this answers is not in the document]
Comment (Zhangdroid) [reply .github/workflows/ci.yaml]: fixed it :)

**Comment by rankun203** [comment]
Hi, I failed to run it as below:

```bash
yarn install
yarn dev
```

Can you explain a bit about the set up process?

**Comment by Zhangdroid** [comment]
> Hi, I failed to run it as below:
> 
> ```shell
> yarn install
> yarn dev
> ```
> 
> Can you explain a bit about the set up process? Thanks

Sorry @rankun203 ,
It should be on README but too busy to write it recently :(

Because Prisma need create a DB, for the first time you run it, need to run 

```
cd api && yarn run dev:prisma
```

After create a local DB, the next time you can just run `yarn dev` on the root dir to start :)

The `dev` still very rough now, it could be better if create a script to manage the client and server development.

**Comment by makara** [comment]
Thanks @Zhangdroid. This is enough for us to take a decision. Feel free to hack more if you want. We'll get back to you soon.

**Pull request closed without merging**